src/FluentMigrator.Runner.Oracle/Processors/Oracle/OracleProcessorBase.cs

```diff
@@ -265,6 +290,92 @@ public override void Process(PerformDBOperationExpression expression)
             expression.Operation?.Invoke(Connection, Transaction);
         }
 
+        /// <summary>
+        /// Splits a SQL script into individual statements, taking into account Oracle-specific syntax rules.
+        /// </summary>
+        private static List<string> SplitOracleSqlStatements(string sqlScript)
```

Comment:
I just noticed (again) the existence of SemicolonSearcher, SqlBatchParser etc, I should use it, what do you think ?

Reply:
I tried to use it (with the same implementation as SnowflakeBatchParser) but it seems that it matches semicolons inside "ranges", in my case : strings 🫠
I'm too afraid to change it 😨

Reply:
Oh I think I understood ... My string spans multiple lines (I'm in the long string tests for Oracle)

Reply:
> I am not a big fan of the batch parser. sql is a very complicated dialect

Yeah, neither am I, but it still a lot better than the Regex previously used

Reply:
The simplest regex would be if the line ends in a `;\s*$`

Reply:
I finally gave up on using the SqlBatchParser, its handling of ranges line-by-line seems broken

Reply:
my regex hack makes it so long as the last non whitespace character on a line is a `;` , it will treat it as a separator.